scripts/tsconfig.json

```diff
@@ -2,9 +2,19 @@
   "extends": "../tsconfig.base.json",
   "compilerOptions": {
     "lib": ["ES2021"],
-    "module": "preserve",
-    "moduleResolution": "bundler"
+    "module": "nodenext",
+    "moduleResolution": "nodenext"
```

Comment:
Since these scripts are executed directly without transpilation, `bundler` didn't make sense.

Reply:
I believe they are run with [`tsx`](https://github.com/mui/base-ui/blob/5daf2048d3c12b6564fad3623638f0c6fe1205c5/scripts/api-docs-builder/package.json#L14). Which uses `esbuild` under the hood, which resolves in a way compatible with `Bundler`. `nodenext` is stricter, so it's definitely not wrong, but if we only run this with `tsx`, then `bundler` should be sufficient.

Reply:
I removed all `tsx` with `node` invocations and everything worked without issues (except I had to replace all `.js` extensions with `.ts` in `scripts/api-docs-builder`. I just had to configure eslint to allow `.ts` imports.
So we can also remove all `tsx` cli calls as well. If others are onboard, I can do that as well.